File: Data Structures & Algorithms/duplicate-integer/submission-0.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        unordered_set<int> ele;
        for (int i = 0 ; i < nums.size(); i ++){
            auto it = ele.find(nums[i]);
            if(it != ele.end()){
                return true;
            }else{
                ele.insert(nums[i]);
            }
        }

        return false;
    }
};

File: Data Structures & Algorithms/find-duplicate-integer/submission-0.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        slow , fast = 0, 0
        while True:
            slow = nums[slow]
            fast = nums[nums[fast]]
            if slow == fast:
                break

        s=0
        while True:
            s = nums[s]
            slow = nums[slow]
            if s == slow:
                return slow